ocrd_calamari/recognize.py

```diff
@@ -54,17 +62,30 @@ def process(self):
             pcgts = page_from_file(self.workspace.download_file(input_file))
 
             page = pcgts.get_Page()
-            page_image, page_xywh, page_image_info = self.workspace.image_from_page(page, page_id)
+            page_image, page_coords, page_image_info = self.workspace.image_from_page(
+                page, page_id, feature_selector=self.features)
 
-            for region in pcgts.get_Page().get_TextRegion():
-                region_image, region_xywh = self.workspace.image_from_segment(region, page_image, page_xywh)
+            for region in page.get_TextRegion():
+                region_image, region_coords = self.workspace.image_from_segment(
+                    region, page_image, page_coords, feature_selector=self.features)
 
                 textlines = region.get_TextLine()
                 log.info("About to recognize %i lines of region '%s'", len(textlines), region.id)
-                for (line_no, line) in enumerate(textlines):
-                    log.debug("Recognizing line '%s' in region '%s'", line_no, region.id)
-
-                    line_image, line_xywh = self.workspace.image_from_segment(line, region_image, region_xywh)
+                for line in textlines:
+                    log.debug("Recognizing line '%s' in region '%s'", line.id, region.id)
```

Comment:
Good idea to change that to output the line id! (I'd prefer something like that in a separate PR though, but in this case I'll let it slip in 😀)